IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
```

Comment:
Yep, ideally, not.

In this case, we have 

`internal func createEntity(json: JSON)`

A proper interface for a domain service should be:

`internal func createEntity(entity: T)`

And the same happens with persistence class.. instead of:

`func add(json: JSON, entityName: String)`

we should have

`func add( entity: T, entityName: String)`

JSON usage should be contained within `RectClient` related classes. Once a response is received, it's parsed into an entity or DTO, and then handled by persistence or domain services.

I'm not sure how difficult that could be with CoreData, do you think we can change it?